dao/src/main/java/org/thingsboard/server/dao/sql/device/DeviceProfileRepository.java

```diff
@@ -57,4 +56,14 @@
 
     DeviceProfileEntity findByTenantIdAndName(UUID id, String profileName);
 
+    @Query(value = "SELECT d.* FROM device_profile as d " +
+            "WHERE d.name = :profileName " +
+            "AND d.profile_data->'configuration'->>'provisionDeviceKey' IS NOT NULL " +
+            "AND d.profile_data->'configuration'->>'provisionDeviceSecret' IS NOT NULL " +
+            "AND d.profile_data->'configuration'->>'provisionDeviceKey' = :provisionDeviceKey " +
```

Comment:
if this query is slow, we should replace embedded json fields with separate columns in the device_profle table.